service/common/src/main/java/org/apache/polaris/service/catalog/io/FileIOFactory.java

```diff
@@ -51,7 +51,7 @@ public interface FileIOFactory {
    * @return a configured FileIO instance.
    */
   FileIO loadFileIO(
-      @Nonnull RealmContext realmContext,
+      @Nonnull CallContext callContext,
```

Comment:
This exposes `PolarisMetaStoreSession` (via `PolarisCallContext`). Do FileIO factories actually need `PolarisCallContext`?

Reply:
What part of `CallContext` is actually required by FileIO factories?

Reply:
to clarify: we're exposing extra information here. Removing what has been exposed is much harder than exposing more details later.

Reply:
Yeah it actually is already needed because it uses the `getOrGenerateSubScopeCreds` which for now is still in the main persistence interface (it's starting to be separated out via the `PolarisCredentialVendor` interface, but that still needs more work to fully split, and either way the Transacting impls need the MetaStoreSession to get the subscoped creds).

So it actually needs *all* of `PolarisCallContext` right now (and before this PR was accessing it via the ThreadLocal).